UPGRADING.md

```diff
@@ -1,5 +1,37 @@
-Upgrading Strava-Ruby-Client
-============================
+# Upgrading Strava-Ruby-Client
+
+### Upgrading to >= 0.5.0
+
+Every Model was upgraded with the possibility to check the API ratelimit that was send as a response from the Strava API.
+
+```ruby
+{
+  limit: limit,
+  usage: usage,
+  total_day: total_day,
+  total_day_usage: total_day_usage,
+  total_day_remaining: total_day_remaining,
+  fiveteen_minutes: fiveteen_minutes, # limit for 15 minute range
+  fiveteen_minutes_usage: fiveteen_minutes_usage, # used past 15 minutes
+  fiveteen_minutes_remaining: fiveteen_minutes_remaining,
+}
+```
+
+Every API call returning a single instance and _not multiple instances_, like `client.athlete` will respond to `#http_response`  
+`Strava::Models::Athlete#http_response`
+
+‼️ Please bear in mind, that you might stumble and fall, when trying to access `http_response` and `ratelimit` info for `activity_comments` being an  
+`Array[Strava::Models::Comments]`!
+
+```ruby
+comments = client.activity_comments(id: 123_456_789)
+
+# comments == Array<Strava::Models::Comment>
+comments.http_response.ratelimit.limit #=> NoMethodError
+
+comment = comments.first
+comment.http_response.ratelimit.limit unless comment.nil?
```

Comment:
This is just new documentation and doesn't belong to UPGRADING (belongs in README). What are the changes I need to make to go to this version in my application? I believe those are:

1. Interface changes where all models now inherit from Response vs. Model. Nothing to do for a user.
2. The methods Strava::Web::Request#get, post, etc. now return a Response, so anyone using those must add `.response` to those. This is an interface change.
3. `::Strava::Web::Response::RaiseError` was renamed.